flow/src/lib.rs

```diff
@@ -627,8 +631,8 @@ pub mod pallet {
 		///
 		/// Emits `CampaignUpdated` event when successful.
 		///
-		/// Weight:
-		#[pallet::weight(1_000_000)]
+		/// Weight: O(1)
+		#[pallet::weight(T::WeightInfo::update_state())]
```

Comment:
This is not constant time, but O(log N) complexity because of the binary search across `CampaignsByState`.
`CampaignsByState` was not limited so far, so N is not defined and we can't calculate weight properly.

As a workaround, we can add a partition by org and limit campaigns by org: double map `CampaignsByOrgByState` and `MaxCampaignsByOrg`.
@2075 @vasylenko-yevhen  WDYT?

In the Control pallet, a similar problem about `OrgMembers` binary search was solved with `MaxMembersPerDAO`.

In the Signal pallet it's still not fixed - `ProposalVoters` binary search. We should limit voters by proposal or change the approach.

Reply:
lgtm, just limit it to a reasonable amount, like 64 campaigns or so. binary search itself is pretty efficient, can understand why you want to have bounds.

Reply:
@2075 just to clarify what should be done.
Remove `CampaignsByState`, use `CampaignsByOrgByState` instead, `MaxCampaignsByOrg` = 64.

What about `ProposalVoters`?